torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2246,38 +2265,57 @@ Index::getReferenceRootPredicates(
     }
 
     // Use the iteration domains extent unless there's a halo extent
-    auto extent = kir_contig_id->extent();
+    kir::Val* start = ir_builder.zeroVal();
+    kir::Val* stop = kir_contig_id->extent();
 
+    // TODO: This isn't used for now. When the consumer has halo,
+    // ShiftPredicateInserter is used.
```

Comment:
Yes, I think that'd be better. Originally, the shift predicate inserter wasn't that much different from the normal path. It was a sort of a thin extension that adds lower-end predicates. However, since then, the normal path is extended with the reference-based approach, and the shift predicate hasn't been updated yet. I don't see any fundamental difficulty to merge them.